### PR TITLE
ci(jenkins): report bundlesize as a GitHub comment

### PR DESCRIPTION
**Pull request by v1v**
Create markdown file on the fly and populate the content as a GitHub comment. See https://github.com/elastic/apm-agent-rum-js/pull/826#issuecomment-650255846

## Issues

Closes https://github.com/elastic/apm-agent-rum-js/issues/667

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env groovy
 
-@Library('apm@current') _
+@Library('apm@feature/bundlesize-step-1') _
 
 pipeline {
   agent { label 'linux && immutable' }
```
Comment (v1v) [inline Jenkinsfile:6]: for testing purposes

File: Jenkinsfile [hunk 2]
```diff
@@ -9,7 +9,7 @@ pipeline {
     BASE_DIR = "src/github.com/elastic/${env.REPO}"
     NOTIFY_TO = credentials('notify-to')
     JOB_GCS_BUCKET = credentials('gcs-bucket')
-    PIPELINE_LOG_LEVEL='INFO'
+    PIPELINE_LOG_LEVEL = 'DEBUG'
     CODECOV_SECRET = 'secret/apm-team/ci/apm-agent-rum-codecov'
     SAUCELABS_SECRET_CORE = 'secret/apm-team/ci/apm-agent-rum-saucelabs@elastic/apm-rum-core'
     SAUCELABS_SECRET = 'secret/apm-team/ci/apm-agent-rum-saucelabs@elastic/apm-rum'
```
Comment (v1v) [inline Jenkinsfile:15]: for testing purposes

File: Jenkinsfile [hunk 3]
```diff
@@ -364,7 +368,8 @@ pipeline {
   }
   post {
     cleanup {
-      notifyBuildResult()
+      // bundlesize id was generated previously, see the bundlesize function
```
Comment (v1v) [inline Jenkinsfile:371]:
```suggestion
      // bundlesize id was generated previously, see the lint stage
```

File: Jenkinsfile [hunk 4]
```diff
@@ -9,7 +9,7 @@ pipeline {
     BASE_DIR = "src/github.com/elastic/${env.REPO}"
     NOTIFY_TO = credentials('notify-to')
     JOB_GCS_BUCKET = credentials('gcs-bucket')
-    PIPELINE_LOG_LEVEL='INFO'
+    PIPELINE_LOG_LEVEL = 'DEBUG'
```
Comment (vigneshshanmugam) [inline Jenkinsfile:12]: Is this still required?

Comment (v1v) [reply Jenkinsfile:12]: Evil GH issues... for some reason it did not push my changes

```
apm-agent-rum-js git:(feature/report-bundlesize) git --no-pager log -1       
commit a990a52150719ba7b1b4d6c64e00249e0e58dfa9 (HEAD -> feature/report-bundlesize, origin/feature/report-bundlesize)
Author: Victor Martinez <VictorMartinezRubio@gmail.com>
Date:   Mon Jun 29 10:59:06 2020 +0100

    ci: use current branch and change comment
```

Comment (v1v) [reply Jenkinsfile]: I've just had to amend my last commit and force push

**Comment by v1v** [comment]
> One minor comment, can we convert the bytes to KB with decimal points? that's a bit easier to read.
Also, it seems the diff part isn't showing anything on this PR, is that because the diff is zero? maybe we should show that as well

I'll enable the KB format in a follow up, since it requires changes in the jenkins shared library.

> it seems the diff part isn't showing anything on this PR,

Good point, this will change as soon as this PR gets merged, then there will be something to compare with, at the moment this PR is the only one with the bundlesize reporting

**Comment by vigneshshanmugam** [comment]
@v1v Not able to see the diff in this PR - https://github.com/elastic/apm-agent-rum-js/pull/825#issuecomment-651127304

**Comment by v1v** [comment]
> @v1v Not able to see the diff in this PR - #825 (comment)

I think the master branch was not yet available with the report -> see [here](https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-rum%2Fapm-agent-rum-mbp/detail/PR-825/6/pipeline/97#step-137-log-1)

Since [there is already a json file](https://apm-ci.elastic.co/job/apm-agent-rum/job/apm-agent-rum-mbp/job/master/lastSuccessfulBuild/artifact/bundlesize.json/*view*/) in the master build then you should see the comment as expected in the next build.

I'll rebuild your PR to see if it behaves as expected

**Pull request merged**